ftd/src/ast/component.rs

```diff
@@ -441,51 +447,102 @@ impl Loop {
 
         let loop_statement = loop_header.value.string(doc_id)?;
 
-        let (on, alias) = ftd::ast::utils::split_at(loop_statement.as_str(), ftd::ast::utils::AS);
+        if loop_header.key.eq("for") {
+            let (alias, on) =
+                ftd::ast::utils::split_at(loop_statement.as_str(), ftd::ast::utils::IN);
 
-        if !on.starts_with(ftd::ast::utils::REFERENCE) && !on.starts_with(ftd::ast::utils::CLONE) {
-            return ftd::ast::parse_error(
-                format!(
-                    "Loop should be on some reference, found: `{}`. Help: use `${}` instead",
-                    on, on
+            let on = if let Some(on) = on {
+                on
+            } else {
+                return ftd::ast::parse_error(
+                    "Statement \"for\" needs a list to operate on",
+                    doc_id,
+                    loop_header.line_number,
+                );
+            };
+
+            if !on.starts_with(ftd::ast::utils::REFERENCE)
+                && !on.starts_with(ftd::ast::utils::CLONE)
+            {
+                return ftd::ast::parse_error(
+                    format!(
+                        "Loop should be on some reference, found: `{}`. Help: use `${}` instead",
+                        on, on
+                    ),
+                    doc_id,
+                    loop_header.line_number,
+                );
```

Comment:
Copying same logic twice just to get `alias` and `on` differently is not a good idea. Merge the logic and use condition to get `alias` and `on`.